iothub/device/samples/Getting Started/SimulatedDevice/Program.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+// This application uses the Azure IoT Hub device SDK for .NET
+// For samples see: https://github.com/Azure/azure-iot-sdk-csharp/tree/main/iothub/device/samples
+
+using System;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Devices.Client;
+using CommandLine;
+
+namespace SimulatedDevice
+{
+    /// <summary>
+    /// This sample illustrates the very basics of a device app sending telemetry. For a more comprehensive device app sample, please see
+    /// <see href="https://github.com/Azure-Samples/azure-iot-samples-csharp/tree/main/iot-hub/Samples/device/DeviceReconnectionSample"/>.
+    /// </summary>
+    internal class Program
+    {
+        //private static Parameters s_parameters;
+        private static DeviceClient s_deviceClient;
+
+        private static async Task Main(string[] args)
+        {
+            Parameters parameters = null;
+            ParserResult<Parameters> result = Parser.Default.ParseArguments<Parameters>(args)
+                .WithParsed(parsedParams => parameters = parsedParams)
+                .WithNotParsed(errors => Environment.Exit(1));
+
+            Console.WriteLine("IoT Hub Quickstarts #1 - Simulated device.");
+
+            // Connect to the IoT hub using the MQTT protocol by default
+            s_deviceClient = DeviceClient.CreateFromConnectionString(parameters.DeviceConnectionString, parameters.TransportType);
+
+            // Set up a condition to quit the sample
+            Console.WriteLine("Press control-C to exit.");
+            using var cts = new CancellationTokenSource();
+            Console.CancelKeyPress += (sender, eventArgs) =>
+            {
+                eventArgs.Cancel = true;
+                cts.Cancel();
+                Console.WriteLine("Exiting...");
+            };
+
+            // Run the telemetry loop
+            await SendDeviceToCloudMessagesAsync(cts.Token);
+
+            // SendDeviceToCloudMessagesAsync is designed to run until cancellation has been explicitly requested by Console.CancelKeyPress.
+            // As a result, by the time the control reaches the call to close the device client, the cancellation token source would
+            // have already had cancellation requested.
+            // Hence, if you want to pass a cancellation token to any subsequent calls, a new token needs to be generated.
+            // For device client APIs, you can also call them without a cancellation token, which will set a default
+            // cancellation timeout of 4 minutes: https://github.com/Azure/azure-iot-sdk-csharp/blob/64f6e9f24371bc40ab3ec7a8b8accbfb537f0fe1/iothub/device/src/InternalClient.cs#L1922
+            await s_deviceClient.CloseAsync();
+
+            s_deviceClient.Dispose();
```

Comment:
Same as above. We can make this a local variable and use the `using` keyword instead.